### PR TITLE
Add "async-await hits beta" blog post

### PR DESCRIPTION
**Pull request by nikomatsakis**
cc @rust-lang/core 

Also, @cramertj and @withoutboats, please give this a once over for technical inaccuracies. =)

### CODE REVIEW

File: posts/2019-09-27-Async-await-hits-beta.md [hunk 1]
```diff
@@ -0,0 +1,160 @@
...
+
+### Restructuring Async I/O in the Rust org
+
+Now that async-await is stable, we are taking the opportunity to make
+some changes to our Rust team structure. The current structure
+includes two working groups: the [Async Foundations WG], focused on
+building up core language support, and the [Async Ecosystem WG],
+focused on supporting the ecosystem develop. 
+
+**In light of all the activity going on in the ecosystem, however,
```
Comment (nikomatsakis) [inline posts/2019-09-27-Async-await-hits-beta.md]: Reminder: I had planned to open an issue on the Governance WG discussing WG teardown procedures and cite that issue here.

File: posts/2019-09-27-Async-await-hits-beta.md [hunk 2]
```diff
@@ -0,0 +1,160 @@
...
+(If you're not familiar with what async-await is, don't despair!
+There's a primer and other details later on in this post!)
+
+### Async-await support in the ecosystem growing rapidly
+
+But async-await has never been the entire story. To make good use of
+async-await, you also need strong libraries and a vibrant ecosystem.
+**Fortunately, you've got a lot of good choices, and they keep getting
+better:** 
+
```
Comment (cramertj) [inline posts/2019-09-27-Async-await-hits-beta.md]: Probably futures-rs should be mentioned in here, too.

Comment (cramertj) [reply posts/2019-09-27-Async-await-hits-beta.md]: Oh, and `hyper` of course.

File: posts/2019-09-27-Async-await-hits-beta.md [hunk 3]
```diff
@@ -0,0 +1,160 @@
...
+The other difference between Rust futures and futures in other
+languages is that they are based on a "poll" model, which makes them
+**zero cost**. In other languages, invoking an async function
+immediately creates a future and schedules it for execution: awaiting
+the future isn't really necessary for it to execute. But this implies
+some overhead for each future that is created. 
+
+In contrast, in Rust, calling an async function does not do any
+scheduling in and of itself, which means that we can compose a complex
+nest of futures without incurring a "per-future cost". As an end-user,
```
Comment (cramertj) [inline posts/2019-09-27-Async-await-hits-beta.md]: nit: no quotes necessary

File: posts/2019-09-27-Async-await-hits-beta.md [hunk 4]
```diff
@@ -0,0 +1,160 @@
...
+though, the main thing you'll notice is that **futures feel "lazy"**:
+they don't do anything until you await them.
+
+If you'd like a closer look at how futures work under the hood,
```
Comment (cramertj) [inline posts/2019-09-27-Async-await-hits-beta.md]: Link also to the "executor" section of the async book, perhaps?

**Comment by cramertj** [comment]
LGTM, thanks!

File: posts/2019-09-27-Async-await-hits-beta.md (hunk 2 above)
Comment (nikomatsakis) [reply posts/2019-09-27-Async-await-hits-beta.md]: @cramertj ok! Have there been any recent developments to cite in particular?

File: posts/2019-09-27-Async-await-hits-beta.md (hunk 1 above)
Comment (nikomatsakis) [reply posts/2019-09-27-Async-await-hits-beta.md]: Done

File: posts/2019-09-27-Async-await-hits-beta.md (hunk 2 above)
Comment (nikomatsakis) [reply posts/2019-09-27-Async-await-hits-beta.md]: For hyper, I see https://github.com/hyperium/hyper/issues/1805

File: posts/2019-09-27-Async-await-hits-beta.md [hunk 5]
```diff
@@ -0,0 +1,173 @@
...
+  start on the new async-await syntax;
+- using [wasm-bindgen-futures], you can even bridge Rust Futures with
+  [JavaScript promises];
+- the [hyper library][hyper] has [migrated][hyper#1805] to adopt standard Rust futures;
+- finally, in addition to the core runtimes just mentioned,
+  async-await support is starting to become available in higher-level
+  [web frameworks][wf] as well.
+  
+[wasm-bindgen-futures]: https://docs.rs/crate/wasm-bindgen-futures/0.2.16
+[tokio]: https://actix.rs/
```
Comment (carllerche) [inline posts/2019-09-27-Async-await-hits-beta.md]: This link is not correct.

File: posts/2019-09-27-Async-await-hits-beta.md [hunk 6]
```diff
@@ -0,0 +1,173 @@
...
+
+### Async-await support in the ecosystem growing rapidly
+
+But async-await has never been the entire story. To make good use of
+async-await, you also need strong libraries and a vibrant ecosystem.
+**Fortunately, you've got a lot of good choices, and they keep getting
+better:** 
+
+- the async runtime [tokio], for example, recently announced an [alpha
+  release][] that supports async-await;
```
Comment (carllerche) [inline posts/2019-09-27-Async-await-hits-beta.md]: I would go further and say "based" on async-await.

**Comment by withoutboats** [comment]
lgtm :+1:

File: posts/2019-09-27-Async-await-hits-beta.md [hunk 7]
```diff
@@ -0,0 +1,173 @@
...
+[async-std]: https://async.rs/
+[recently announced]: https://async.rs/blog/announcing-async-std/
+[wf]: https://www.arewewebyet.org/topics/frameworks/
+[JavaScript promises]: https://developer.mozilla.org/en-US/docs/Web/JavaScript/Guide/Using_promises
+[hyper]: https://hyper.rs
+[hyper#1805]: https://github.com/hyperium/hyper/issues/1805
+
+### Restructuring Async I/O in the Rust org
+
+Now that async-await is stable, we are taking the opportunity to make
```
Comment (jplatte) [inline posts/2019-09-27-Async-await-hits-beta.md]: Is "stable" being used here to mean something other than "available on the stable channel"? This seems a bit confusing.

File: posts/2019-09-27-Async-await-hits-beta.md [hunk 8]
```diff
@@ -0,0 +1,173 @@
...
+
+**The [Async Foundations WG], meanwhile, will continue, but with a
+shift in focus.** Now that async-await is en route to stabilization,
+the focus will be on polish, such as improving diagnostics, fixing
+smaller bugs, and improving documentation such as the [async
+book]. Once progress is made on that, we'll be considering what
+features to implement next.
+
+(An aside: this is the first time that we've ever opted to spin *down*
+a working group before, and we realized that we don't have a formal
```
Comment (jplatte) [inline posts/2019-09-27-Async-await-hits-beta.md]: 'before' probably is from an earlier revision of this sentence?

File: posts/2019-09-27-Async-await-hits-beta.md (hunk 7 above)
Comment (nikomatsakis) [reply posts/2019-09-27-Async-await-hits-beta.md]: changed to "approaching stable"

Comment (nikomatsakis) [reply posts/2019-09-27-Async-await-hits-beta.md]: changed to "approaching stable"

File: posts/2019-09-27-Async-await-hits-beta.md (hunk 2 above)
Comment (cramertj) [reply posts/2019-09-27-Async-await-hits-beta.md]: WRT futures? no, just pointing people at the futures-preview 0.3.0-alpha and giving a heads up that 0.3.0 will arrive alongside stable async/await.

File: posts/2019-09-27-Async-await-hits-beta.md [hunk 9]
```diff
@@ -0,0 +1,173 @@
...
+
+If you'd like a closer look at how futures work under the hood, take a
+look at [the executor section] of the [async book], or watch the
+[excellent talk][video] that [withoutboats] gave at [Rust LATAM 2019]
+on the topic.
+
+[the executor section]: https://rust-lang.github.io/async-book/02_execution/04_executor.html
+[video]: https://www.youtube.com/watch?v=skos4B5x7qE
+[Rust LATAM 2019]: https://rustlatam.org/
+[withoutboats]: https://rustlatam.org/
```
Comment (lqd) [inline posts/2019-09-27-Async-await-hits-beta.md]:
```suggestion
[withoutboats]: https://github.com/withoutboats
```

Comment (nikomatsakis) [reply posts/2019-09-27-Async-await-hits-beta.md]: good catch :)

**Pull request merged**